src/common/notifications.cpp

```diff
@@ -0,0 +1,85 @@
+#include "pch.h"
+#include "notifications.h"
+
+#include <unknwn.h>
+#include <winrt/base.h>
+#include <winrt/Windows.Foundation.h>
+#include <winrt/Windows.Data.Xml.Dom.h>
+#include <winrt/Windows.Foundation.Collections.h>
+#include <winrt/Windows.UI.Notifications.h>
+#include <winrt/Windows.ApplicationModel.Background.h>
+
+using namespace winrt::Windows::ApplicationModel::Background;
+using winrt::Windows::Data::Xml::Dom::XmlDocument;
+using winrt::Windows::UI::Notifications::ToastNotification;
+using winrt::Windows::UI::Notifications::ToastNotificationManager;
+
+namespace
+{
+    constexpr std::wstring_view TASK_NAME = L"PowerToysBackgroundNotificationsHandler";
+    constexpr std::wstring_view TASK_ENTRYPOINT = L"PowerToysNotifications.BackgroundHandler";
+}
+
+void notifications::register_background_toast_handler()
+{
+    try
+    {
+        // Re-request access to clean up from our previous versions
+        BackgroundExecutionManager::RemoveAccess();
+        BackgroundExecutionManager::RequestAccessAsync().get();
+
+        BackgroundTaskBuilder builder;
+        ToastNotificationActionTrigger trigger{ L"PowerToys" };
```

Comment:
We should never hard-code string, not even the product name (it may have to be localized in some languages).